pkg/fmap/fmap_test.go

```diff
@@ -120,7 +120,7 @@ func TestNoSignature(t *testing.T) {
 	fakeFlash := bytes.Repeat([]byte{0x53, 0x11, 0x34, 0x22}, 94387)
 	r := bytes.NewReader(fakeFlash)
 	_, _, err := Read(r)
-	expected := "Cannot find fmap signature"
```

Comment:
This is our fault, but you may as well fix it. Rather than comparing error strings, compare the error itself:

```go
var errSigNotFound = errors.New("cannot find FMAP signature")

func Read(...) ... {

  return errSigNotFound
}
```

and in the test
```go
_, _, got := Read(r)
if want := errSigNotFound; got != want {
  t.Errorf("Read(%v) = %v, want %v", r, got, want)
}
```

Please use `want` instead of `expected`, and place what you got before what you want as in the example. Don't do "expected: %s, got: %s". (We do it wrong below.)

Covered also by https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

Reply:
Many thanks for the suggestions. It's my first contribution to a go project.